src/userspace/exec_monitor.c

```diff
@@ -0,0 +1,70 @@
+/*
```

Comment:
I see that this file is compiled in C. This should be C++ because you will soon need to handle C++ types such as the vector of std::string.